jujugui/static/gui/src/app/components/inspector-change-version/inspector-change-version.js

```diff
@@ -68,13 +68,40 @@ YUI.add('inspector-change-version', function() {
     },
 
     /**
-    The callable to be passed to the version item to change versions.
+      The callable to be passed to the version item to change versions.
 
-      @method _versionButtonAction
-      @param {String} charmId The charm id.
-      @param {Object} e The click event.
+        @method _versionButtonAction
+        @param {String} charmId The charm id.
+        @param {Object} e The click event.
     */
     _versionButtonAction: function(charmId, e) {
+      // Fetch the macaroon if one is available so that they can add private
+      // charms to their models.
+      var macaroon = this.props.getMacaroon();
+      // In Juju 2.0+ we have to add the charm before we set it. This is just
+      // good practice for 1.25.
+      this.props.addCharm(charmId, macaroon,
+        this._addCharmCallback.bind(this, charmId), {
+          // XXX hatch: the ecs doesn't yet support addCharm so we are going to
```

Comment:
We don't :) Basically we rely on the fakebackends implementation of loadCharm to fetch the charm data. https://github.com/juju/juju-gui/blob/develop/jujugui/static/gui/src/app/store/env/fakebackend.js#L329 so it only actually works as a 'side effect' which is why we've had to mark things as 'loaded' manually